.github/workflows/build_wheels.yml

```diff
@@ -9,24 +9,46 @@ on:
   pull_request:
     paths:
       - 'pixi.lock'
-      - '.github/workflows/release.yml'
       - '.github/workflows/_build_*_pip.yml'
+      - '.github/workflows/build_wheels.yml'
 
 jobs:
+  get_version:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          # For setuptools_scm to be able to determine version,
+          # we need to fetch the entire history
+          fetch-depth: 0
+      - name: Get Version
+        id: get_version
+        run: |
+          set -exo pipefail
+          pip install setuptools_scm
+          echo "bodo_version=`python -m setuptools_scm`" >> $GITHUB_OUTPUT
+    outputs:
+      bodo_version: ${{ steps.get_version.outputs.bodo_version }}
```

Comment:
We ran into a problem where cibuildwheel was using the wrong version during build. This fixes that by fetching the version and fixing it beforehand